graylog2-web-interface/src/components/sidecars/configurations/CopyConfigurationModal.jsx

```diff
@@ -43,47 +42,37 @@ class CopyConfigurationModal extends React.Component {
   };
 
   _save = () => {
-    const configuration = this.state;
+    const { copyConfiguration } = this.props;
+    const { errorMessage, id, name } = this.state;
 
-    if (!configuration.error) {
-      this.props.copyConfiguration(this.state.id, this.state.name, this._saved);
+    if (!errorMessage) {
+      copyConfiguration(id, name, this._saved);
     }
   };
 
   _changeName = (event) => {
-    const nextName = event.target.value;
-    this.setState({ name: nextName });
-    this.props.validateConfiguration({ name: nextName }).then((validation) => {
-      let errorMessage = '';
+    const { validateConfiguration } = this.props;
+    const name = event.target.value;
+    this.setState({ name, errorMessage: undefined });
+
+    validateConfiguration({ name }).then((validation) => {
       if (validation.errors.name) {
-        errorMessage = validation.errors.name[0];
+        this.setState({ errorMessage: validation.errors.name[0] });
       }
-      this.setState({ error: validation.failed, error_message: errorMessage });
     });
   };
 
   render() {
+    const { errorMessage, name } = this.state;
+
     return (
-      <span>
-        <MenuItem className={style.menuItem} onSelect={this.openModal}>Clone</MenuItem>
-        <BootstrapModalForm ref="modal"
-                            title="Clone"
-                            onSubmitForm={this._save}
-                            submitButtonDisabled={this.state.error}
-                            submitButtonText="Done">
-          <fieldset>
-            <Input type="text"
-                   id={this._getId('configuration-name')}
-                   label="Name"
-                   defaultValue={this.state.name}
-                   onChange={this._changeName}
-                   bsStyle={this.state.error ? 'error' : null}
-                   help={this.state.error ? this.state.error_message : 'Type a name for the new configuration'}
-                   autoFocus
-                   required />
-          </fieldset>
-        </BootstrapModalForm>
-      </span>
+      <CloneMenuItem onSelect={this.openModal}
+                     onSave={this._save}
+                     id={this._getId('configuration-name')}
+                     onChange={this._changeName}
+                     error={errorMessage}
```

Comment:
I saw this in my browser console, I think this prop needs a default value here or make the prop not required:

```
Warning: Failed prop type: The prop `error` is marked as required in `CloneMenuItem`, but its value is `undefined`.
    in CloneMenuItem (created by CopyConfigurationModal)
    in CopyConfigurationModal (created by ConfigurationRow)
    in ul (created by DropdownMenu)
    in RootCloseWrapper (created by DropdownMenu)
    in DropdownMenu (created by Dropdown)
```